OpenSim/Moco/MocoGoal/MocoOutputGoal.h

```diff
@@ -72,11 +88,30 @@ class OSIMMOCO_API MocoOutputGoal : public MocoGoal {
             "false)");
     OpenSim_DECLARE_PROPERTY(divide_by_mass, bool,
             "Divide by the model's total mass (default: false)");
+    OpenSim_DECLARE_PROPERTY(exponent, int,
+            "The exponent applied to the output value in the integrand "
+            "(default: 1).");
+    OpenSim_DECLARE_PROPERTY(output_index, int,
+            "The index to the value to be minimized when a vector type "
+            "Output is specified. For SpatialVec Outputs, indices 0, 1, "
+            "and 2 refer to the rotational components and indices 3, 4, "
+            "and 5 refer to the translational components. A value of -1 "
+            "indicates to minimize the vector norm (default: -1).");
```

Comment:
I understand this "when a vector types" documentation, but can we actually enforce it?